src/modules/job-info/job_state.c

```diff
@@ -13,6 +13,10 @@
 #if HAVE_CONFIG_H
 #include "config.h"
 #endif
+#ifndef _GNU_SOURCE
```

Comment:
_GNU_SOURCE is already unconditionally defined in `configure.ac`.

Reply:
Doh!  My comment was lost apparently, but I was concerned about this comment in the man page
> with glibc, one gets the POSIX version of basename() when <libgen.h> is
       included, and the GNU version otherwise.

One might include that file later accidentally, or even transitively.  Will badness result?

Since it's so simple, maybe better to avoid the situation and provide a little helper function that looks for trailing slash with strrchr.

Reply:
Sounds like a good idea.  I thought there were corner cases other than this (such as if the user specified `..`), but trailing slash seems to be it.  Can simply tweak the glibc implementation.

```
char *
__basename (const char *filename)
{
  char *p = strrchr (filename, '/');
  return p ? p + 1 : (char *) filename;
}
```